x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview_logic.ts

```diff
@@ -0,0 +1,130 @@
+  apiLogsUnavailable: boolean;
+  documentCount: number;
+  startDate: string;
+  endDate: string;
+  operationsPerDay: number[];
+  queriesPerDay: number[];
+  totalClicks: number;
+  totalQueries: number;
+}
+interface EngineOverviewValues extends EngineOverviewApiData {
+  dataLoading: boolean;
+  timeoutId: number | null;
+}
+
+interface EngineOverviewActions {
+  setPolledData(engineMetrics: EngineOverviewApiData): EngineOverviewApiData;
+  setTimeoutId(timeoutId: number): { timeoutId: number };
+  pollForOverviewMetrics(): void;
+  onPollingSuccess(engineMetrics: EngineOverviewApiData): EngineOverviewApiData;
+}
+
+export const EngineOverviewLogic = kea<MakeLogicType<EngineOverviewValues, EngineOverviewActions>>({
+  path: ['enterprise_search', 'app_search', 'engine_overview_logic'],
+  actions: () => ({
+    setPolledData: (engineMetrics) => engineMetrics,
+    setTimeoutId: (timeoutId) => ({ timeoutId }),
+    pollForOverviewMetrics: true,
+    onPollingSuccess: (engineMetrics) => engineMetrics,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        setPolledData: () => false,
+      },
+    ],
+    apiLogsUnavailable: [
+      false,
+      {
+        setPolledData: (_, { apiLogsUnavailable }) => apiLogsUnavailable,
+      },
+    ],
+    startDate: [
+      '',
+      {
+        setPolledData: (_, { startDate }) => startDate,
+      },
+    ],
+    endDate: [
+      '',
+      {
+        setPolledData: (_, { endDate }) => endDate,
+      },
+    ],
+    queriesPerDay: [
+      [],
+      {
+        setPolledData: (_, { queriesPerDay }) => queriesPerDay,
+      },
+    ],
+    operationsPerDay: [
+      [],
+      {
+        setPolledData: (_, { operationsPerDay }) => operationsPerDay,
+      },
+    ],
+    totalQueries: [
+      0,
+      {
+        setPolledData: (_, { totalQueries }) => totalQueries,
+      },
+    ],
+    totalClicks: [
+      0,
+      {
+        setPolledData: (_, { totalClicks }) => totalClicks,
+      },
+    ],
+    documentCount: [
+      0,
+      {
+        setPolledData: (_, { documentCount }) => documentCount,
+      },
+    ],
+    timeoutId: [
+      null,
+      {
+        setTimeoutId: (_, { timeoutId }) => timeoutId,
+      },
+    ],
+  }),
+  listeners: ({ actions }) => ({
+    pollForOverviewMetrics: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.get(`/api/app_search/engines/${engineName}/overview`);
+        actions.onPollingSuccess(response);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    onPollingSuccess: (engineMetrics) => {
+      const timeoutId = window.setTimeout(actions.pollForOverviewMetrics, POLLING_DURATION);
+      actions.setTimeoutId(timeoutId);
+      actions.setPolledData(engineMetrics);
+    },
+  }),
+  events: ({ values }) => ({
+    beforeUnmount() {
+      if (values.timeoutId !== null) clearTimeout(values.timeoutId);
```

Comment:
I saw a post about nullish operators/assignment on Twitter recently and almost kinda wanted to try it out here:

```suggestion
      values.timeoutId ?? clearTimeout(values.timeoutId);
```

But it felt kinda weird to read and Typescript didn't like it. haha

Reply:
I wonder why TS didn't like it?

One interesting thing...

In the former, if `values.timeoutId` is `undefined`, `clearTimeout` would be called.

In the later, if `values.timeoutId` is `undefined`, `clearTimeout` would NOT be called.

Reply:
Looking at the code now, I realize that is irrelevant, because timeoutId would never be null

Reply:
TS probably isn't yet fully updated for nullish operators is my guess.

> timeoutId would never be null

I think you meant `undefined` here? If so, correct, a redux value can never be `undefined` or the entire app just crashes

Reply:
There's often a large disconnect between my brain and my fingers 😬